packages/odata-common/src/uri-conversion/get-resource-path.ts

```diff
@@ -108,7 +108,7 @@ export function createGetResourcePathForKeys(
     schema: Record<string, any>
   ): string {
     const edmType = schema[upperCaseSnakeCase(key)].edmType;
-    return `${key}=${uriConverter(value, edmType)}`;
+    return `${key}=${encodeURIComponent(uriConverter(value, edmType))}`;
```

Comment:
You encode now all parameters which makes sense I guess. However, on the de-serializer level there is also the [serializeToUri](https://github.com/SAP/cloud-sdk-js/blob/701002de058d658c2204517e065cabef8eb13d9c/packages/odata-common/src/de-serializers/default-de-serializers.ts#L102) property. I wonder where the encoding should go. There are arguments in both direction:
- If it goes to the serialize, users can overwrite it.
- If we have it here we ensure all properties are encoded

Perhaps the colleagues @marikaner @florian-richter @jjtang1985 @tomfrenken can also give their opinion. No matter where we put it, it should be then either here or in the de-serializer.

Reply:
@florian-richter , @FrankEssenberger : The URI serialize methods in de-serializers were also my first thought, but that resulted in filter expressions being encoded twice (linked [PR](https://github.com/SAP/cloud-sdk-js/pull/1283)). I think if we want to use serializeToUri (or similar) methods , somehow changes from this PR will have to be adjusted. Please correct me if my understanding is wrong.

Regarding encoding the keys, this [adr](https://github.com/SAP/cloud-sdk-js/blob/main/knowledge-base/adr/0024-url-encoding.md) mentions that we assume we don't need to do it unless we get feedback from users, so I left it out.

Reply:
I got lost a bit.
I feel we need some diagrams so we can clarify all these layers.
I see this is a refactoring ticket and we need to very the test coverage.

Reply:
So I have been thinking about this issue and I think it is correct the way it is done to put the uri compoent rather high in the get-ref-path method. Doing so it is analogous to the `get-filter.ts`. Also note the important detail, that the encodeURICompoent does not touch `'`. This is only relevant for the value of the filter or key, because the valid odata String is `key='myvalue'`. If the myvalue contains `'` these need to be encoded which is done my the [serializeToUri](https://github.com/SAP/cloud-sdk-js/blob/701002de058d658c2204517e065cabef8eb13d9c/packages/odata-common/src/de-serializers/default-de-serializers.ts#L102).

Reply:
Deeksha will add a comment that one remembers this detail and also two more tests to cover the `/` and `'` isssue.